### DISCUSSION TITLE
How to render a specific ray through a scene

### DISCUSSION THREAD

**Discussion by chinmay0301ucsd** [mitsuba-renderer/mitsuba3#395] [Q&A]
Hi, 
Is it possible to render a specific ray through a scene using mitsuba API? Basically I want to use mi.render(scene), but but on a set of randomly generated rays instead of using the sensor directly. What would be a good way to do this?

**Comment by Speierers** [comment]
Yes, this is possible. I would recommend you take a look at [this tutorial](https://mitsuba.readthedocs.io/en/stable/src/rendering/scripting_renderer.html).

Otherwise, you can also use the `Integrator.sample()` method directly, providing your own rays.
